luna/gateware/usb/usb2/endpoints/isochronous_stream_in.py

```diff
@@ -0,0 +1,220 @@
+
+    Used for repeatedly streaming data to a host from a stream-like interface.
+    Intended to be useful as a transport for e.g. video or audio data.
+
+    Attributes
+    ----------
+    stream: amaranth.lib.stream.Interface, input stream
+        Full-featured stream interface that carries the data we'll transmit to the host.
+
+    interface: EndpointInterface
+        Communications link to our USB core.
+
+    data_requested: Signal(), output
+        Strobes, when a new packet starts
+
+    frame_finished: Signal(), output
+        Strobes immediately after the last byte in a frame has been transmitted
+
+    bytes_in_frame: Signal(range(0, 3073)), input
+        Specifies how many bytes will be transferred during this frame. If this is 0,
+        a single ZLP will be emitted; for any other value one, two, or three packets
+        will be generated, depending on the packet size. Latched in at the start of
+        each frame.
+
+        The maximum allowed value for this signal depends on the number of transfers
+        per (micro)frame:
+        - If this is a high-speed, high-throughput endpoint (descriptor indicates
+          maxPacketSize > 512 and multiple transfers per microframe), then this value
+          maxes out at (N * maxPacketSize), where N is the number of transfers per microframe.
+        - For all other configurations, this must be <= the maximum packet size.
+
+    Parameters
+    ----------
+    endpoint_number: int
+        The endpoint number (not address) this endpoint should respond to.
+    max_packet_size: int
+        The maximum packet size for this endpoint. Should match the wMaxPacketSize provided in the
+        USB endpoint descriptor.
+    """
+
+    _MAX_FRAME_DATA = 1024 * 3
+
+    def __init__(self, *, endpoint_number, max_packet_size):
+
+        self._endpoint_number = endpoint_number
+        self._max_packet_size = max_packet_size
+
+        #
+        # I/O Port
+        #
+        self.interface      = EndpointInterface()
+        self.stream         = stream.Interface(stream.Signature(unsigned(8)))
+        self.data_requested = Signal()
+        self.frame_finished = Signal()
+
+        self.bytes_in_frame = Signal(range(0, self._MAX_FRAME_DATA + 1))
+
+    def elaborate(self, platform):
+        m = Module()
+
+        # Shortcuts.
+        interface        = self.interface
+        tx_stream        = interface.tx
+        new_frame        = interface.tokenizer.new_frame
+
+        targeting_ep_num = (interface.tokenizer.endpoint == self._endpoint_number)
+        targeting_us     = targeting_ep_num & interface.tokenizer.is_in
+        data_requested   = targeting_us & interface.tokenizer.ready_for_response
+
+        # Track our transmission state.
+        bytes_left_in_frame  = Signal.like(self.bytes_in_frame)
+        bytes_left_in_packet = Signal(range(0, self._max_packet_size + 1), reset=self._max_packet_size - 1)
```

Comment:
I suggest replacing`reset=` for `init=`, as done in #278

Reply:
Tx! Fixed in: 8e248df